tests/functional/sca/test_scan_ci.py

```diff
@@ -0,0 +1,86 @@
+from pathlib import Path
+
+from tests.functional.utils import run_ggshield_sca_scan
+from tests.repository import Repository
+
+
+def test_scan_ci_diff(tmp_path: Path, monkeypatch, pipfile_lock_with_vuln) -> None:
+    """
+    GIVEN a repository with a commit containing a vuln,
+    two clean commits on top of it, and a CI env
+    WHEN scanning the last two commits, it's OK
+    WHEN scanning the commit containing the vuln
+    THEN the vuln is found
+    """
+    repo = Repository.create(tmp_path)
+    repo.create_commit()
+
+    secret_file = repo.path / "Pipfile.lock"
+    secret_file.write_text(pipfile_lock_with_vuln)
+    repo.add("Pipfile.lock")
+
+    for _ in range(3):
+        repo.create_commit()
+
+    env = {"CI": "true", "GITLAB_CI": "true"}
+    for key, value in env.items():
+        monkeypatch.setenv(key, value)
+
+    #
+    monkeypatch.setenv("CI_COMMIT_BEFORE_SHA", "HEAD~1")
+    run_ggshield_sca_scan("ci", expected_code=0, cwd=repo.path)
+
+    monkeypatch.setenv("CI_COMMIT_BEFORE_SHA", "HEAD~2")
```

Comment:
what if `CI_COMMIT_BEFORE_SHA` is equal to `EMPTY_SHA`? When I look at our gitlab CI it's very often equal to this value

Reply:
Here I am just using one way of specifying the commits for the tests, but multiple ways, including the one where `CI_COMMIT_BEFORE_SHA` is `EMPTY_SHA`, are handled in `gitlab_ci_range`. This is tested so it shouldn't be a problem.